Example/SmileID/DocumentVerification/DocumentVerificationWithSelector.swift

```diff
@@ -0,0 +1,43 @@
+import SmileID
+import SwiftUI
+
+struct DocumentVerificationWithSelector: View {
+    @State private var verificationDetails: DocumentVerificationDetails?
+
+    let userId: String
+    let jobId: String
+    let delegate: DocumentVerificationResultDelegate
+
+    init(
+        userId: String,
+        jobId: String,
+        delegate: DocumentVerificationResultDelegate
+    ) {
+        self.userId = userId
+        self.jobId = jobId
+        self.delegate = delegate
+    }
+
+    var body: some View {
+        VStack {
+            if let verificationDetails {
```

Comment:
is this the best way to do this? using an if to check if the state of this struct changes or something like that

Just curious but since this is the sample app, we can continue with it

Reply:
That was how the code was written initially. I only extracted the properties into single struct. Also this is kind of unwrapping the optional property `verificationDetails` if it's nil, it will show the other view not that I'm using it to check the state. The state drives the view refresh not the other way around. In SwiftUI we use if statement to show/hide a view since the `.hidden()` modifier doesn't take a boolean.